test/integration/default/inspec/default_spec.rb

```diff
@@ -26,14 +26,20 @@
 end
 
 describe file('/etc/vault/vault.json') do
-  its('mode') { should eq 0640 }
   it { should be_file }
   it { should be_owned_by 'vault' }
   it { should be_grouped_into 'vault' }
+  its('mode') { should cmp '0640' }
 end
 
 describe service('vault') do
   it { should be_installed }
   it { should be_enabled }
   it { should be_running }
 end
+
+describe file('/usr/local/bin/vault') do
+  it { should exist }
+  it { should be_symlink }
+  it { should be_linked_to '/opt/vault/0.7.2/vault' }
```

Comment:
All of the `0.7.2` values in this file should be variable-ized, but I'm not sure how to do that with custom resources.